### DISCUSSION TITLE
GitHub Copilot free for teacher

### DISCUSSION THREAD

**Discussion by joseluisgs** [community/community#19837] [GitHub Education] [answered]
Hi

My name is Jose Luis Gonzalez. I'm a GitHub Campus Advisor, and Phd in Software Development and Teracher of software Developmen in Spain.

I would like to make a request and it is if it is possible to study that the professors and campus advisor can access CoPilot with our boxfor free.

For me, as a software development teacher, it is essential because it saves me a lot of time.

As an advisor and teacher  I would like to be able to continue using it and teaching with it in class because I think that it is very useful for various tasks and that students understand how and when to use it.

I find it very interesting that since students can access to it, development teachers and GitHub advisors can access it to help and recommend how to use it to other teachers, classmates, students and companies we work with every day.

I would ask you to study that option, for me it would be essential and many teachers would like it.

Thank you very much
José Luis González
https://github.com/joseluisgs

**Comment by themiralay** [comment]
Thanks for share us Mr Jose

**Reply by rodo1985** [reply]
I am in the same situation of Jose Luis and I would like to support his query. It would be great for teachers and for our students as well because we can do class teaching the use of the copilot that it is great to save time while developing.

**Reply by joseluisgs** [reply]
Thank you, but  we don't have any answer for GitHub Education support

**Reply by kbzowski** [reply]
I strongly support the opinion of my predecessors. It is hard to talk about the capabilities of the software without access to it. I think this opinion will be supported by more teachers in a similar situation when they notice that the trial period is over and they cannot continue to use it.

**Reply by ashmanmalhotra** [reply]
I support it too.

**Reply by TeroKeso** [reply]
Minimal level of support would be a company level support using Microsoft 365 licencing framework. For Education institutions it is hard just "use credit card" type of purchase

**Reply by ashmanmalhotra** [reply]
> Minimal level of support would be a company level support using Microsoft 365 licencing framework. For Education institutions it is hard just "use credit card" type of purchase

Do you mean it is free with Microsoft 365? If yes then how?

**Reply by luc122c** [reply]
> > Minimal level of support would be a company level support using Microsoft 365 licencing framework. For Education institutions it is hard just "use credit card" type of purchase
> 
> Do you mean it is free with Microsoft 365? If yes then how?

No, I believe what he's saying is that in educational settings it's usually difficult to pay via card and that there should be alternate payment options like Microsoft 365.

**Reply by TeroKeso** [reply]
> > > Minimal level of support would be a company level support using Microsoft 365 licencing framework. For Education institutions it is hard just "use credit card" type of purchase
> > 
> > 
> > Do you mean it is free with Microsoft 365? If yes then how?
> 
> No, I believe what he's saying is that in educational settings it's usually difficult to pay via card and that there should be alternate payment options like Microsoft 365.

Microsoft has a established licensing framework [https://github.com/AaronDinnage/Licensing](https://github.com/AaronDinnage/Licensing) It compicated, but in use in many companyes and Instutution. Would be nice if GitHub Copilot would piggyback on that.

**Reply by joseluisgs** [reply]
Please share this post 
We dont hace any answer

**Reply by iamgoodbytes** [reply]
In the same boat as you all. I've taught my students about co-pilot and they now use it in my classes and even on their exams, but I can't use it anymore as a teacher. Thinking about banning it for that reason, which would be ridiculous as well.

**Reply by anthonymobile** [reply]
+1 from Cornell. i'm cutting CoPilot from fall syllabus as this is not resolved.

**Reply by joseluisgs** [reply]
Pelease share. We don't have any answer.

**Reply by Nembreizh** [reply]
+1

**Reply by javiermilesinacap** [reply]
It's frustrating for us, who will promote these tools. +2

**Reply by TedSwanseaUni** [reply]
I'm very surprised teachers don't get free access to co-pilot. I know my students want to use it, but I can't show them how or advice them if I can't access it myself.  I don't want to pay out my own pocket when coding is just one part of what we teach. I hope Microsoft reconsider this.

**Reply by henrikhjelmse** [reply]
i can not speak to the class about copilot. 
if i don't have access to it or can use it.

**Reply by nbourre** [reply]
I'm in the same boat!

**Reply by faustort** [reply]
Totally agree with this, as a teacher knowing new technologies help us build a better community.

**Reply by luiscunhacsc** [reply]
I am also a Teacher and just received the message that my copilot "trial" had ended, just before the beginning of the semester. Please do as all other companies providing educational licenses: teachers and students are part of the same community. Thank you.

**Reply by crmdevtool** [reply]
good!

**Reply by alanmartinc** [reply]
Good!!

**Comment by kbuc** [comment]
It would be great if this was included in Github Education.

**Reply by ZOSTEV3** [reply]
This would improve the site a whole lot better! I support the idea.

**Comment by heremansgeert** [comment]
Same here! Would be great for teachers also. The educational budget isn't everywhere the same around the world. Here in Belgium our High Schools have to spend every penny twice.

**Comment by divais** [comment]
We have heard you and are actively working on this and will come back shortly 🤞🏼
Thank you for your patience.

**Reply by Erzambayu** [reply]
thank you for your consideration, as a student I think this is also necessary to help teachers understand better what their students are asking for

**Reply by melvincwng** [reply]
Glad to see GitHub's generosity in allowing both students & teachers to use their amazing products for free!

Cheers 🎉🥳

**Reply by luiscunhacsc** [reply]
Thank you, @divais . I am also a teacher and I will be looking forward to check the developments on this policy. It is strange, because copilot is the only product providing educational/student licensing that differentiates between students and their teachers (at least from the basket of selected tools I have been using for the last years). Thanks in advance.

**Reply by joseluisgs** [reply]
Thanks a lot. Teachers appreciate these initiatives. Many of us work in public institutions with a limited budget and many times we cannot justify this type of license.
In addition, as campus advisor we can advise colleagues and students if we all have the same opportunities.

**Reply by gugawag** [reply]
Thank you. We are looking forward this 😉

**Reply by joseluisgs** [reply]
Yes, please give us news soon 🤞

**Reply by LinusLing** [reply]
new semester has just begun. Look forward to this.

**Reply by btvd** [reply]
Same here! It will be really nice to have this option for teachers!

**Reply by hello-amed** [reply]
This would be super helpful, cheers!!

**Reply by phandinhlong** [reply]
Thank you. I look forward to using it soon.

**Reply by fafulin** [reply]
Same here!

**Comment by henrikhjelmse** [comment]
update ?

**Reply by joseluisgs** [reply]
No news yet

**Reply by luiscunhacsc** [reply]
Thank you for starting this conversation, Jose Luis.

**Reply by joseluisgs** [reply]
Much appreciated.
It is difficult to communicate, but I think we all agree on the need to be able to use CoPilot as part of the educational package for teachers.

**Reply by TeroKeso** [reply]
> Much appreciated. It is difficult to communicate, but I think we all agree on the need to be able to use CoPilot as part of the educational package for teachers.

Or Microsoft 365 Education model

https://docs.microsoft.com/en-us/office365/servicedescriptions/office-365-platform-service-description/microsoft-365-education

**Reply by joseluisgs** [reply]
I think it is better to have the teacher's toolbox as the students, https://education.github.com/teachers. I'm also a github campus advisor, I'm supporting students and teachers on my campus. I'm working on CoPilot questions, but I don't have access to CoPilot. Sometimes some decisions are difficult to make sense of. We are all educational community.

**Comment by luiscunhacsc** [comment]
I am also waiting patiently for their reply, but my semester is starting in a week, and I really needed to know if I will be using Copilot for my classes' demos or not (I record new videos every semester, using the most up-to-date tools, and I wish I could use Copilot, but the policy towards teachers reveals lack of sensitivity from Copilot to what is expected from Educational Licenses (and what is the current practice followed by the vast majority of other big companies). Thank you. Any timely update would be welcomed, @divais . I am waiting all that I can before I start recording my videos (which I do using free tools such as OBS and OpenShot, after happily quitting paying the prices Camtasia wanted for their "Educational" license.

**Reply by joseluisgs** [reply]
Mee too

**Reply by fafulin** [reply]
Mee too

**Reply by divais** [reply]
Thank you for your patience, [Copilot is now available free for verified teachers](https://github.blog/2022-09-08-github-copilot-now-available-for-teachers/). If you’re interested in sharing your experience, feedback, or ideas, please [join the conversation](https://github.com/community/Global-Campus-Teachers/discussions/118) with other verified teachers on GitHub Global Campus.

**Reply by TedSwanseaUni** [reply]
Thanks @divais, this will be super helpful for many of us.

**Comment by fafulin** [comment]
anything news?

**Reply by joseluisgs** [reply]
no :)

**Comment by divais** [comment, marked as answer]
Thank you for your patience, [Copilot is now available free for verified teachers](https://github.blog/2022-09-08-github-copilot-now-available-for-teachers/). If you’re interested in sharing your experience, feedback, or ideas, please [join the conversation](https://github.com/community/Global-Campus-Teachers/discussions/118) with other verified teachers on [GitHub Global Campus](https://education.github.com/globalcampus/teacher).

**Reply by joseluisgs** [reply]
Yes. ♥️♥️♥️♥️

**Reply by joseluisgs** [reply]
Thank you so muxh

**Reply by iamgoodbytes** [reply]
Fantastic news. I have to add that the development community actively working in the education field really appreciates it when a company like Github/Microsoft acknowledges and listens to the problems teachers often encounter. Thanks a lot for not ignoring these questions and for taking such timely action, right before the start of the new school year! 🫡

**Reply by divais** [reply]
We are here for you and would love to learn your experience and feedback.

**Reply by elmaestrotic** [reply]
Many thanks to GitHub for providing Copilot. It's really good for us as teachers of programming to have tools like this, good luck!

**Comment by pitosalas** [comment]
I am a verified teacher. But when I go to copilot it asks me to start a free trial or pay. How do I get the free access to co-pilot?

**Reply by kbuc** [reply]
> I am a verified teacher. But when I go to copilot it asks me to start a free trial or pay. How do I get the free access to co-pilot?

@pitosalas 
Try going to the GitHub "Settings" and the select "Github Copilot" on the left panel. Then click "Enable GitHub Copilot" button.

**Reply by pitosalas** [reply]
Worked! Thank you!

**Reply by almansa** [reply]
Hi, this solution didn't work for me. I am a verified teacher.
As suggested by @kbuc I went to my GitHub "Settings" and then select "Github Copilot" on the left panel. But then I do not get the "Enable GitHub Copilot" button. Instead I only get a "Try copilot for 60 days" button.

I was last verified as a faculty on Sep 9, 2021. Should I re-verify my academic affiliation to enable Copilot as a teacher?

**Reply by kbuc** [reply]
@almansa You should check what kind of plan you have under Settings / Billing and plans / Plans and usage. There you will see your current plan and also your eligibility for copilot under Add-ons.

**Reply by almansa** [reply]
Worked! Thank you @kbuc !

**Comment by irem6142** [comment]


**Reply by divais** [reply]
hi :)

**Comment by Ashish-Abraham** [comment]
Would be great if Copilot was free for student verified accounts too.😊

**Comment by alimttth** [comment]
I am also a Teacher and just received the message that my copilot "trial" had ended, just before the beginning of the semester. Please do as all other companies providing educational licenses: teachers and students are part of the same community. Thank you.

**Comment by TedSwanseaUni** [comment]
GitHub have made Copilot free for teachers now so this issue should now be closed.

**Comment by ismailertuglu14** [comment]
Thanks

**Comment by hombit** [comment]
I'm a verified teacher (July 2022 last time), but do not have an access to the free Copilot. "Settings -> Copilot -> Activate" asks to select a plan and confirm my credit card info

**Reply by yanyuechuixue** [reply]
Me too, i am a teacher in physics at a Chinese university. My access to GitHub copilot has been disabled today.

**Comment by yanyuechuixue** [comment]
I am a teacher in physics at a Chinese university. My access to GitHub copilot has been disabled today.

It shows I have to subscribe to use GitHub copilot. Is this program has end?

**Reply by yanyuechuixue** [reply]
Just go https://github.com/github-copilot/free_signup and click. Then the access comes back.
Thanks all.

**Reply by hombit** [reply]
It doesn't work to me =(

**Reply by timosachsenberg** [reply]
same problem here...

**Comment by igsbma** [comment]
Thank you for making Copilot available for Teacher.

I am a verified University of Maryland faculty as of Mar 8, 2023. However, under Setting --> Copilot, it still asked me to try 60 days free. Please help! Attached is the screenshot.
<img width="1164" alt="Screenshot 2023-03-10 at 11 04 22 AM" src="https://user-images.githubusercontent.com/1796954/224365270-dd1d461f-44d8-4fa3-a511-07fb97dbd255.png">

**Reply by samus-aran** [reply]
Hi @igsbma 👋 

You should be presented with the ability to "Enable" Copilot. But if this isn't available try https://github.com/github-copilot/free_signup to confirm your status as a verified GitHub Global Campus Teacher.

Let us know how you get on 👍

**Reply by igsbma** [reply]
@samus-aran Thank you for the reply! **Your link worked!** I am in now.

**Reply by timosachsenberg** [reply]
hmm doesn't work for me

**Reply by chenweiguang2017** [reply]
> Hi @igsbma 👋
> 
> You should be presented with the ability to "Enable" Copilot. But if this isn't available try https://github.com/github-copilot/free_signup to confirm your status as a verified GitHub Global Campus Teacher.
> 
> Let us know how you get on 👍

Great! It's works for me. Thanks a lot.

**Comment by WJC-POG** [comment]
I've applied almost 20 times but keep getting rejections. I've tried uploading my work documents, course schedule, and employment contract. I am no longer sure what other official documents I need to provide to prove that I am a university teacher.
Does anyone have a solution?

**Reply by Easy-Ez** [reply]
Avoid using vpn, use campus network for application, preferably use edge browser and enable locate permission.
It is better to use a photo instead of uploading, and you can use a white paper to write your name, school and github id.
Try again. haha

**Reply by javiermilesinacap** [reply]
And if you use a translator on page, it neither work.

**Comment by erdemerbaba** [comment]
Thank you for your contribution
